src/code/audio/audio_load.c

```diff
@@ -1,161 +1,2110 @@
+        case OS_TV_PAL:
+            gAudioContext.unk_2960 = 20.03042f;
+            gAudioContext.refreshRate = 50;
+            break;
+        case OS_TV_MPAL:
+            gAudioContext.unk_2960 = 16.546f;
+            gAudioContext.refreshRate = 60;
+            break;
+        case OS_TV_NTSC:
+        default:
+            gAudioContext.unk_2960 = 16.713f;
+            gAudioContext.refreshRate = 60;
+    }
+
+    Audio_InitMesgQueues();
+
+    for (i = 0; i < ARRAY_COUNT(gAudioContext.aiBufLengths); i++) {
+        gAudioContext.aiBufLengths[i] = 0xA0;
+    }
+
+    gAudioContext.totalTaskCnt = 0;
+    gAudioContext.rspTaskIdx = 0;
+    gAudioContext.curAIBufIdx = 0;
+    gAudioContext.soundMode = AUDIO_MODE_STEREO;
+    gAudioContext.currTask = NULL;
+    gAudioContext.rspTask[0].task.t.dataSize = 0;
+    gAudioContext.rspTask[1].task.t.dataSize = 0;
+    osCreateMesgQueue(&gAudioContext.syncDmaQueue, &gAudioContext.syncDmaMesg, 1);
+    osCreateMesgQueue(&gAudioContext.currAudioFrameDmaQueue, gAudioContext.currAudioFrameDmaMesgBuf,
+                      ARRAY_COUNT(gAudioContext.currAudioFrameDmaMesgBuf));
+    osCreateMesgQueue(&gAudioContext.externalLoadQueue, gAudioContext.externalLoadMesgBuf,
+                      ARRAY_COUNT(gAudioContext.externalLoadMesgBuf));
+    osCreateMesgQueue(&gAudioContext.preloadSampleQueue, gAudioContext.preloadSampleMesgBuf,
+                      ARRAY_COUNT(gAudioContext.preloadSampleMesgBuf));
+    gAudioContext.curAudioFrameDmaCount = 0;
+    gAudioContext.sampleDmaCount = 0;
+    gAudioContext.cartHandle = osCartRomInit();
+
+    if (heap == NULL) {
+        gAudioContext.audioHeap = gAudioHeap;
+        gAudioContext.audioHeapSize = gAudioContextInitSizes.heapSize;
+    } else {
+        void** hp = &heap;
+        gAudioContext.audioHeap = *hp;
+        gAudioContext.audioHeapSize = heapSize;
+    }
+
+    for (i = 0; i < (s32)gAudioContext.audioHeapSize / 8; i++) {
+        ((u64*)gAudioContext.audioHeap)[i] = 0;
+    }
+
+    AudioHeap_InitMainPools(gAudioContextInitSizes.initPoolSize);
+
+    for (i = 0; i < ARRAY_COUNT(gAudioContext.aiBuffers); i++) {
+        gAudioContext.aiBuffers[i] = AudioHeap_AllocZeroed(&gAudioContext.audioInitPool, AIBUF_LEN * sizeof(s16));
+    }
+
+    gAudioContext.sequenceTable = (AudioTable*)gSequenceTable;
+    gAudioContext.soundFontTable = (AudioTable*)gSoundFontTable;
+    gAudioContext.sampleBankTable = (AudioTable*)gSampleBankTable;
+    gAudioContext.sequenceFontTable = gSequenceFontTable;
+    gAudioContext.numSequences = gAudioContext.sequenceTable->numEntries;
+
+    gAudioContext.audioResetSpecIdToLoad = 0;
+    gAudioContext.resetStatus = 1;
+
+    AudioHeap_ResetStep();
+    AudioLoad_InitTable(gAudioContext.sequenceTable, _AudioseqSegmentRomStart, 0);
+    AudioLoad_InitTable(gAudioContext.soundFontTable, _AudiobankSegmentRomStart, 0);
+    AudioLoad_InitTable(gAudioContext.sampleBankTable, _AudiotableSegmentRomStart, 0);
```

Comment:
I personally prefer the macro `SEGMENT_ROM_START` in cases like this
```suggestion
    AudioLoad_InitTable(gAudioContext.sequenceTable, SEGMENT_ROM_START(Audioseq), 0);
    AudioLoad_InitTable(gAudioContext.soundFontTable, SEGMENT_ROM_START(Audiobank), 0);
    AudioLoad_InitTable(gAudioContext.sampleBankTable, SEGMENT_ROM_START(Audiotable), 0);
```

Reply:
Nice, thanks